### DISCUSSION TITLE
NetworkObject - add hooks for better host visibility handling?

### DISCUSSION THREAD

**Discussion by cameronoltmann** [FirstGearGames/FishNet#334] [Feature Request]
As mentioned [here](https://github.com/FirstGearGames/FishNet/discussions/211), broader support for managing NetworkObject clientHost visibility would be nice.  What's in place already is good, but has a two specific drawbacks:

- Renderers are handled nicely, but other components like lights and audiosources aren't
- Procgen scenarios where child objects are instantiated after initial NOB spawning aren't properly supported.

Ideally I'd love to see:

- NetworkObject._lastClientHostVisibility exposed in some fashion (or something similar - basically a way to query "is this object supposed to be visible now?" - mostly useful on init)
- Hooks for UpdateRendererVisibility so that I can control things like audiosources, lights, etc

I saw mention of possibly exposing this in the above discussion - did that end up going anywhere?

**Comment by FirstGearGames** [comment]
Added in 3.7.8
- Added NetworkObject.OnHostVisibilityUpdated.
